source/common/http/conn_manager_utility.cc

```diff
@@ -107,8 +110,17 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
         Utility::appendXff(request_headers, *connection.remoteAddress());
       }
     }
-    request_headers.insertForwardedProto().value().setReference(
-        connection.ssl() ? Headers::get().SchemeValues.Https : Headers::get().SchemeValues.Http);
+    if (trusted_forwarded_proto) {
```

Comment:
nit: I think you can just fold this in below as `!trusted_forwarded_proto || ...` right?

Reply:
I can, though I lean towards not.
In practice I find it easier to verify there's no behavioral change if you have
if (guard) {
  new_behavior
} else {
  old_behavior
}
It's also easier to clean up guarded behavior by just removing an if else block.  This is simple enough I can go either way but I kind of prefer using our internal best practices for safe addition/removal of code since they evolved from a bunch of bugs where we had booleans inverted and accidentally turned things on by default and off when we flipped flags etc.

Reply:
Sure that's fine.